### PR TITLE
[12.0] Add wizard to bind shopinvader.partner

### PR DESCRIPTION
**Pull request by lmignon**
forward port of #323
 [ADD] Add wizard to bind shopinvader.partner (same behaviour like partner portal access rights)

### CODE REVIEW

File: shopinvader/wizards/shopinvader_partner_binding.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        :return:
+        """
+        if self.env.context.get("active_model") == "res.partner":
+            partner_ids = self.env.context.get("active_ids", [])
+            lines = [(6, False, [])]
+            for partner in self.env["res.partner"].browse(partner_ids):
+                shopinv_partner = partner.shopinvader_bind_ids.filtered(
+                    lambda x, b=self.shopinvader_backend_id: x.backend_id == b
+                )
+                if shopinv_partner:
```
Comment (rousseldenis) [inline shopinvader/wizards/shopinvader_partner_binding.py:57]: For better readability I'd prefer (I don't like 'continue''s):

```suggestion
                shopinvader_partners = self.env["res.partner"].browse(partner_ids).shopinvader_bind_ids.filtered(
                    lambda x, b=self.shopinvader_backend_id: x.backend_id == b)
                for shopinvader_partner in shopinvader_partners:
```

but it's nittpicking

File: shopinvader/wizards/shopinvader_partner_binding_line.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    )
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Partner",
+        required=True,
+        ondelete="cascade",
+        readonly=True,
+    )
+    email = fields.Char(
+        related="partner_id.email", required=True, readonly=True
```
Comment (rousseldenis) [inline shopinvader/wizards/shopinvader_partner_binding_line.py:28]: In v12, readonly is not necessary

File: shopinvader/wizards/shopinvader_partner_binding_line.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            backend = (
+                record.shopinvader_partner_binding_id.shopinvader_backend_id
+            )
+            # Ensure the binding doesn't exist yet
+            partner_binding = record.partner_id.shopinvader_bind_ids.filtered(
+                lambda x, b=backend: x.backend_id == b
+            )
+            if not partner_binding:
+                bind_values = {"backend_id": backend.id}
+                record.partner_id.write(
```
Comment (rousseldenis) [inline shopinvader/wizards/shopinvader_partner_binding_line.py:57]: Shouldn't these writes be grouped by value to avoid multiple ones?

**Review by rousseldenis — APPROVED** [review]
Code review. Two questions

File: shopinvader/wizards/shopinvader_partner_binding.py (hunk 1 above)
Comment (lmignon) [reply shopinvader/wizards/shopinvader_partner_binding.py:57]: I prefer continue. Moreover you only have 1 binding by partner by backend...

File: shopinvader/wizards/shopinvader_partner_binding_line.py (hunk 3 above)
Comment (lmignon) [reply shopinvader/wizards/shopinvader_partner_binding_line.py:57]: not possible.... each line is for a different partner

Comment (rousseldenis) [reply shopinvader/wizards/shopinvader_partner_binding_line.py:57]: OK, l meant partners grouped by backend

**Pull request merged**